diff --git a/src/3rd_party/sentencepiece/src/CMakeLists.txt b/src/3rd_party/sentencepiece/src/CMakeLists.txt
index 0ba6407..276caec 100644
--- a/ src/3rd_party/sentencepiece/src/CMakeLists.txt
+++ b/src/3rd_party/sentencepiece/src/CMakeLists.txt
@@ -159,6 +159,9 @@ set(SPM_TEST_SRCS
 find_package(Threads REQUIRED)
 
 set(SPM_LIBS ${PROTOBUF_LITE_LIBRARY} Threads::Threads)
+if(ANDROID)
+    set(SPM_LIBS ${SPM_LIBS} android log)
+endif(ANDROID)
 
 if (SPM_ENABLE_NFKC_COMPILE)
   find_package(ICU 4.4 COMPONENTS i18n data uc REQUIRED)
